dot/parachain/bitfield-distribution/bitfield_distribution.go

```diff
@@ -0,0 +1,533 @@
+	err = vdm.SetValue(message)
+	if err != nil {
+		return err
+	}
+	jobData.onePerValidator[validatorID] = &vdm
+
+	relayMessage(jobData, topology, b.peerViews, validatorID, message, requiredRouting, b.subSystemToOverseer)
+
+	modifyReputation(b.reputation, b.subSystemToOverseer, event.PeerID, util.UnifiedReputationChange{
+		Type:   util.BenefitMinorFirst,
+		Reason: "Valid message with new information",
+	}, relayParent)
+
+	return nil
+}
+
+func (b *BitfieldDistribution) processUpdatedAuthorityIDsEvent(event networkbridgeevents.UpdatedAuthorityIDs) error {
+	//TODO implement in #4360
+	panic("implement me")
+}
+
+func (b *BitfieldDistribution) ProcessActiveLeavesUpdateSignal(signal parachaintypes.ActiveLeavesUpdateSignal) error {
+	//TODO implement in #4361
+	panic("implement me")
+}
+
+func (b *BitfieldDistribution) ProcessBlockFinalizedSignal(signal parachaintypes.BlockFinalizedSignal) error {
+	return nil
+}
+
+func (b *BitfieldDistribution) Stop() {
+	logger.Infof("Stopping BitfieldDistribution subsystem")
+}
+
+// relayMessage distributes a given valid and signature checked bitfield message.
+//
+// Can be originated by another subsystem or received via network from another peer.
+func relayMessage(jobData *perRelayParentData, topologyNeighbors *grid.GridNeighbours, peers map[peer.ID]struct {
+	view            parachaintypes.View
+	protocolVersion uint32 // ignore v1 peers after we have validation protocol version support
+}, validatorID parachaintypes.ValidatorID, message validationprotocol.CheckedBitfield,
+	requiredRouting grid.RequiredRouting, subsystemToOverSeerChan chan<- any) {
+	relayParent := message.Hash
+
+	// notify the overseer about a new and valid signed bitfield
+	go func() {
+		subsystemToOverSeerChan <- provisionermessages.ProvisionableDataBitfield{
+			RelayParent: relayParent,
+			Bitfield:    message.CheckedSignedAvailabilityBitfield,
+		}
+	}()
+
+	// pass on the bitfield distribution to all interested peers
+	// 1. get interested peers
+	interestedPeers := make(map[peer.ID]uint32)
+	for peerID, peerData := range peers {
+		if peerData.view.Contains(relayParent) {
+			if jobData.messageFromValidatorNeededByPeer(peerID, validatorID) {
+				needRouting := topologyNeighbors.ShouldRouteToPeer(requiredRouting, peerID)
+				if needRouting {
+					interestedPeers[peerID] = peerData.protocolVersion
+				}
+			}
+		}
+	}
+
+	if len(interestedPeers) == 0 {
+		logger.Infof("no peers are interested in gossip for relay parent")
+		return
+	}
+
+	// 2. insert the message sent for this peerData into jobData
+	for peerID := range interestedPeers {
+		jobData.messageSentToPeer[peerID] = map[parachaintypes.ValidatorID]struct{}{validatorID: {}}
+	}
+
+	// 3. send NetworkBridgeTxMessage::SendValidationMessage to all v2 and v3 peers
+	v2InterestedPeers := filterByPeerVersion(interestedPeers, 2)
+	if len(v2InterestedPeers) != 0 {
+		go func() {
+			// TODO: add version support for validation protocol
```

Comment:
when will we need this?

Reply:
goroutine will be removed here, unless you are referring to the validation protocol version? which I think it is out of scope and will be implemented in a different PR